<?xml version="1.0"?>

<launch>
	<!--Calibration Parameters-->
  <param name="base_width" type="double" value="0.56"/><!--Robot Base Width-->
  <param name="dcal_L" type="double" value="1.05"/><!--Left Wheel m/rev-->
  <param name="dcal_R" type="double" value="1.05"/><!--Right Wheel m/rev-->
  <param name="ecal_L" type="double" value="4863.0"/><!--Left Encoder C/rev-->
  <param name="ecal_R" type="double" value="4848.0"/><!--Right Encoder C/rev-->
  <param name="KPA" type="double" value="0.1"/><!--Angular Proportional Gain-->
  <param name="KDA" type="double" value="0.001"/><!--Angular Derivative Gain-->
  <param name="KPL" type="double" value="0.1"/><!--Linear Proportional Gain-->
  <param name="KDL" type="double" value="0.0001"/><!--Linear Derivative Gain-->

  <arg name="model"/>
  <arg name="gui" default="False"/>
  <param name="robot_description" command="cat $(find robot_odometry)/URDF/rangerX.urdf"/>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" value="10.0"/>
  </node>
  <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
    <param name="use_gui" value="True"/>
  </node>

  <node pkg="robot_odometry" type="rangerx_odom" name="rangerX_odom" required="false" output="screen"/>

  <node pkg="rosserial_arduino" type="serial_node.py" name="serial_node">
    <param name="~port" value="/dev/ttyACM0"/>
  </node>

  <node pkg="rplidar_ros"  type="rplidarNode" name="rplidarNode">
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>  
    <param name="serial_baudrate"     type="int"    value="256000"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <param name="scan_mode"           type="string" value="Standard"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"/>
    
  <param name="use_sim_time" value="false"/>
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <param name="base_frame" value="base_footprint"/>
    <param name="map_frame" value="map"/>
    <param name="odom_frame" value="odom"/>
    <param name="xmin" value="-50.0"/>
    <param name="ymin" value="-50.0"/>
    <param name="xmax" value="50.0"/>
    <param name="ymax" value="50.0"/>
  </node>

</launch>
